lib/Provider/Command/MessageSend.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Mail\Provider\Command;
+
+use OCA\Mail\Db\LocalMessage;
+use OCA\Mail\Service\AccountService;
+use OCA\Mail\Service\Attachment\AttachmentService;
+use OCA\Mail\Service\OutboxService;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\IConfig;
+use OCP\Mail\Provider\Exception\SendException;
+use OCP\Mail\Provider\IAddress;
+use OCP\Mail\Provider\IMessage;
+
+class MessageSend {
+
+	public function __construct(
+		protected IConfig $config,
+		protected ITimeFactory $time,
+		protected AccountService $accountService,
+		protected OutboxService $outboxService,
+		protected AttachmentService $attachmentService
+	) {
+	}
+
+	/**
+	 * performs send operation
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $userId			system user id
+	 * @param string $serviceId			mail account id
+	 * @param IMessage $message			mail message object with all required parameters to send a message
+	 * @param array $options			array of options reserved for future use
+	 *
+	 * @return LocalMessage
+	 */
+	public function perform(string $userId, string $serviceId, IMessage $message, array $option = []): LocalMessage {
+		// find user mail account details
+		$account = $this->accountService->find($userId, (int) $serviceId);
+		// convert mail provider message to mail app message
+		$localMessage = new LocalMessage();
+		$localMessage->setType($localMessage::TYPE_OUTGOING);
+		$localMessage->setAccountId($account->getId());
+		$localMessage->setSubject($message->getSubject());
+		$localMessage->setBody($message->getBody());
```

Comment:
The local message send fails with:

```json
{"reqId":"go23DSZ0ui2eI6J3jaPr","level":3,"time":"2024-08-20T15:04:22+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"mail","method":"PUT","url":"/index.php/apps/mail/api/outbox/602","message":"OCA\\Mail\\Controller\\OutboxController::update(): Argument #5 ($editorBody) must be of type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 208 in file '/var/www/nextcloud/apps/mail/lib/Controller/OutboxController.php' line 176","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:129.0) Gecko/20100101 Firefox/129.0","version":"31.0.0.1","exception":{"Exception":"Exception","Message":"OCA\\Mail\\Controller\\OutboxController::update(): Argument #5 ($editorBody) must be of type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 208 in file '/var/www/nextcloud/apps/mail/lib/Controller/OutboxController.php' line 176","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":161,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":309,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1003,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":24,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","Line":146,"Previous":{"Exception":"TypeError","Message":"OCA\\Mail\\Controller\\OutboxController::update(): Argument #5 ($editorBody) must be of type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 208","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":208,"function":"update","class":"OCA\\Mail\\Controller\\OutboxController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":114,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":161,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":309,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/lib/base.php","line":1003,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/index.php","line":24,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/nextcloud/apps/mail/lib/Controller/OutboxController.php","Line":176},"message":"OCA\\Mail\\Controller\\OutboxController::update(): Argument #5 ($editorBody) must be of type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 208 in file '/var/www/nextcloud/apps/mail/lib/Controller/OutboxController.php' line 176","exception":{},"CustomMessage":"OCA\\Mail\\Controller\\OutboxController::update(): Argument #5 ($editorBody) must be of type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 208 in file '/var/www/nextcloud/apps/mail/lib/Controller/OutboxController.php' line 176"}}
```

you probably need to set the editorBody as well here.

Reply:
@SebastianKrupinski ^ is this solved?